src/backend/distributed/utils/colocation_utils.c

```diff
@@ -728,12 +719,47 @@ CheckDistributionColumnType(Oid sourceRelationId, Oid targetRelationId)
 }
 
 
+/*
+ * GetColumnTypeEquality checks if distribution column types and collations
+ * of the given columns are same. The function sets the boolean pointers.
+ */
+void
+GetColumnTypeEquality(Var *sourceDistributionColumn, Var *targetDistributionColumn,
```

Comment:
I think we can change this to `CheckColumnTypeEquality`, then we can include the error messages in here as well so we don't need to duplicate them.

Reply:
makes sense